src/lib/gundb/UserStorageClass.js

```diff
@@ -1929,11 +1930,11 @@ export class UserStorage {
     //saving index by onetime code so we can retrieve and update it once withdrawn
     //or skip own withdraw
     if (event.type === EVENT_TYPE_SEND && event.data.code) {
-      const hashedCode = this.wallet.wallet.utils.sha3(event.data.code)
+      const hashedCode = event.data.hashedCode || this.wallet.wallet.utils.sha3(event.data.code)
```

Comment:
1. this code repeats so we can change logic a bit. 
2. There's an constant EVENT_TYPE_WITHDRAW should be used
3. also, we could avoid.long.paths using destructuring
4. we could reuse  Gun's node ref feed.get('codeToTxHash'):

```
const { wallet, feed } = this; 
const { utils } = wallet.wallet;
const { id: eventId, date, type, data } = event;
const { code, hashedCode } = data;

if (code) {
  let ownLink;
  const eventHashedCode = hashedCode || utils.sha3(code);
  const codeToTxHashRef = feed.get('codeToTxHash')
  
  switch(type) {
    case EVENT_TYPE_SEND:
      codeToTxHashRef.put({ [eventHashedCode]: eventId })
      break
    case EVENT_TYPE_WITHDRAW:
      ownLink = await codeToTxHashRef.get(eventHashedCode)
      
      if (!ownLink) {
        break
      }
      
      logger.debug('updateFeedEvent: skipping own link withdraw', { event })

      feed.get(`queue.${eventId}`).put(null)
      return event
    default
      break
  }
}
```